app/storage/storage_encryption.py

```diff
@@ -50,15 +49,16 @@ def encrypt_data(self, data):
             'kid': '1,1',
         }
 
-        jwe_token = jwe.JWE(plaintext=base64url_encode(data), protected=protected_header)
-
-        jwe_token.add_recipient(self.key)
+        jwe_token = jwe.JWE(
+            plaintext=data,
+            protected=protected_header,
+            recipient=self.key,
+        )
 
         return jwe_token.serialize(compact=True)
 
     def decrypt_data(self, encrypted_token):
-
         jwe_token = jwe.JWE(algs=['dir', 'A256GCM'])
         jwe_token.deserialize(encrypted_token, self.key)
 
-        return base64url_decode(jwe_token.payload.decode()).decode()
+        return jwe_token.payload
```

Comment:
For full backwards compatibility, we'd need to worry about the session and submitted response too

Reply:
Ah yeah. I'll add in some compatibility for those storage models too..